go/libkb/base58.go

```diff
@@ -14,10 +14,12 @@ var alphabetMap map[byte]uint8
 
 func getAlphabetMap() map[byte]uint8 {
 	if alphabetMap == nil {
-		alphabetMap = make(map[byte]uint8)
+		newMap := make(map[byte]uint8)
 		for i, c := range []byte(alphabet) {
-			alphabetMap[c] = uint8(i)
+			newMap[c] = uint8(i)
 		}
+		// We do this atomically to avoid needing to take a lock.
```

Comment:
Sorry, I worded that poorly. What I meant was, in this approach each racing thread will initialize it's own separate map, and the only shared piece of state they'll ever touch will be the `alphabetMap` pointer. This should be safe to overwrite, as long as what you're overwriting it with is an identical map, and no writes to these maps ever happen after `alphabetMap` is updated to point to them.

I started with a Mutex, and then I figured this way might be simpler and also faster in the common case. But maybe `once.Do` is just as good and more explicit? Do you have a preference?